open_optical_gating/cli/prospective_optical_gating.py

```diff
@@ -267,31 +313,56 @@ def predict_trigger_wait(
         logger.warning("Phase discrepency, trigger aborted. At {0} with wait {1} for target {2} [{3}]", thisFramePhase%(2*np.pi), phaseToWait, settings["targetSyncPhase"], thisFramePhase - (multiPhaseCounter*2*np.pi))
         timeToWaitInSecs = 0.0
 
+    # This logic catches cases where we are predicting a long way into the future using only a small number of datapoints.
+    # That is likely to be error-prone, so (unless using the "barrier frame" logic) we may increase
+    # the number of frames we use for prediction.
     frameInterval = 1.0 / settings["framerate"]
     if allowedToExtendNumberOfFittedPoints and timeToWaitInSecs > (
         settings["extrapolationFactor"] * settings["minFramesForFit"] * frameInterval
     ):
+        # TODO: JT writes: this approach of editing settings[minFramesForFit] and then changing it back again feels really messy to me.
+        # We should hold off changing it for now, though, because I think we may want to refactor how settings is used
+        # Once that is complete, a better solution here may present itself naturally.
+        # TODO: JT writes: BUG: The recursive function call is also using completely the wrong parameters!!
+        # (This may not have been hit in testing because you are probably using barrier frames, which prevent this code branch from running)
         settings["minFramesForFit"] *= 2
         if (
             settings["minFramesForFit"] <= pastPhases.shape[0]
             and settings["minFramesForFit"] <= settings["maxFramesForFit"]
         ):
             logger.info("Increasing number of frames to use")
-            timeToWaitInSecs = predict_trigger_wait(
-                pastPhases, settings["targetSyncPhase"]
-            )
+            # Recurse, using a larger number of frames, to obtain an improved predicted time
+            timeToWaitInSecs = predict_trigger_wait(       # TODO: JT writes: this function call is passing completely the wrong parameters!!
+                pastPhases, settings["targetSyncPhase"]    # This parent function is also going to do the wrong thing if output=="phases".
+            )                                              # If "phases" is not used anywhere, it is probably best just to remove that option entirely.
         settings["minFramesForFit"] = settings["minFramesForFit"] // 2
 
+    # Return our prediction
     if output == "seconds":
         return timeToWaitInSecs
     elif output == "phases":
         return phaseToWait
     else:
-        logger.critical("What are ye on mate!")
+        logger.critical("What are ye on mate!")   # TODO: JT writes: can you be more helpful here!?
```

Comment:
This is one of those catches for errors that 'should never happen' but one day did. I think, if we remove the phases option we can also remove this catch because we will always use seconds.